### PR TITLE
udp application filter test support

### PR DESCRIPTION
**Pull request by badhrinathpa**
1. Create UDP packet and send to destination. 

2. Add port information in config to send application filter based packet.

3. Current test is incomplete. We would need to check upf and router pod to see if packet has been dropped or transferred.

4. Updated default data packet event to icmp packet event

### CODE REVIEW

File: realue/worker/pdusessworker/handler.go [hunk 1]
```diff
@@ -26,6 +60,188 @@ const (
 	IPV4_MIN_HEADER_LEN int = 20
 )
 
+func checksum(buf []byte) uint16 {
```
Comment (gajjarvini) [inline realue/worker/pdusessworker/handler.go:63]: There is similar function in the util package, should we reuse it ?

File: realue/worker/pdusessworker/handler.go [hunk 2]
```diff
@@ -92,6 +308,12 @@ func SendIcmpEchoRequest(pduSess *realuectx.PduSession) (err error) {
 	return nil
 }
 
+func HandleUdpMessage(pduSess *realuectx.PduSession,
```
Comment (thakurajayL) [inline realue/worker/pdusessworker/handler.go:311]: Are we getting downlink data as well ?

File: realue/worker/pdusessworker/handler.go [hunk 3]
```diff
@@ -154,14 +376,19 @@ func HandleDlMessage(pduSess *realuectx.PduSession,
 		if err != nil {
 			return fmt.Errorf("failed to handle icmp message:%v", err)
 		}
+	case 17:
+		err = HandleUdpMessage(pduSess, dataMsg.Payload[ipv4Hdr.Len:])
+		if err != nil {
+			return fmt.Errorf("failed to handle icmp message:%v", err)
```
Comment (thakurajayL) [inline realue/worker/pdusessworker/handler.go:382]: log change. UDP

File: simue/handler.go [hunk 4]
```diff
@@ -448,15 +448,35 @@ func HandleProcedure(ue *simuectx.SimUe) {
...
-	case common.USER_DATA_PKT_GENERATION_PROCEDURE:
+	case common.UDP_DATA_PKT_GENERATION_PROCEDURE:
+		ue.Log.Infoln("Initiating Udp Data Packet Generation Procedure")
+		msg := &common.UeMessage{}
+		msg.UserDataPktCount = ue.ProfileCtx.DataPktCount
+		if ue.ProfileCtx.DefaultAs == "" {
+			ue.ProfileCtx.DefaultAs = "192.168.250.1" // default destination for AIAB
+		}
+		msg.DefaultAs = ue.ProfileCtx.DefaultAs
+		msg.DefaultAsSrcPort = ue.ProfileCtx.DefaultAsSrcPort
```
Comment (thakurajayL) [inline simue/handler.go:459]: Should not this also be having some default UDP port ?

File: realue/worker/pdusessworker/handler.go (hunk 1 above)
Comment (badhrinathpa) [reply realue/worker/pdusessworker/handler.go]: done

File: simue/handler.go (hunk 4 above)
Comment (badhrinathpa) [reply simue/handler.go:461]: done

**Comment by gab-arrobo** [comment]
@badhrinathpa can you please rebase this PR? Thanks!

**Pull request closed without merging**